### PR TITLE
Move page rendering to the model

### PR DESCRIPTION
**Pull request by encukou**
I think it would be good to make rendering of the lesson pages independent of the Flask app. That way the backend can be reused for CLI/desktop apps, tools, etc.
The Flask app then only provides navigation and non-content pages.

### CODE REVIEW

**Comment by hroncok** [comment]
I feel a collision with https://github.com/pyvec/naucse.python.cz/pull/71

Should I wait?

**Comment by encukou** [comment]
Yeah, there'll need to be a rebase :( 

If you have time, could you do a review for this? I'm not too attached to the last commit, but I guess everything before it is at the point where it can use more eyes.

File: naucse/models.py [hunk 1]
```diff
@@ -145,6 +145,7 @@ def default_lesson_url(lesson, page='index'):
             'subpage_url': lambda page: lesson_url(lesson=lesson, page=page),
             'lesson': lesson,
             'page': self,
+            '$solutions': solutions,
```
Comment (hroncok) [inline naucse/models.py]: Is this some kind of convention I'm not aware of? Why `$`?

**Comment by hroncok** [comment]
1. site freezning works
 2. site appears to look the same
 3. the new render script sucessfully generates HTML fragments
 4. changes make sense generally

The render script generates bogus links to static content and other pages. Since it's experimental, I think it's OK, but want to make sure you are aware of the problem.

File: naucse/models.py (hunk 1 above)
Comment (encukou) [inline naucse/models.py]: It makes the variable unusable from normal code. Could be any character that can't be used in variable names.
I'm not aware of any conventions, I just picked something.

**Comment by encukou** [comment]
I'll merge without the script, then.
I think the script's code could be useful for an Edux exporter, but I don't really want to maintain the script itself now.

**Comment by hroncok** [comment]
> I think the script's code could be useful for an Edux exporter

Generating HTML does not help much, I can do that with current setup and converting HTML to dokuwiki is not easy.

File: naucse/models.py (hunk 1 above)
Comment (hroncok) [inline naucse/models.py]: I might not see why is that useful here. Care to elaborate?

**Comment by encukou** [comment]
Well, it's hopefully easier now to get Jinja-processed Markdown, as well.

**Pull request closed without merging**